src/nendo_plugin_quantize_core/plugin.py

```diff
@@ -75,32 +76,69 @@ def extract_beat(self, y, sr):
 
         return tempo, beat_frames
 
-    def extract_beat_essentia(self, y, sr):
+    def extract_beat_beatnet(self, filename: str):
         """Extracts beat from a given audio signal and converts beats to frames.
 
         Args:
-            y: The audio signal.
-            sr: The sample rate.
+            filename: The name of the target file.
 
         Returns:
             A tuple of (tempo, beat_frames).
         """
-        rhythm_extractor = es.RhythmExtractor2013(method="multifeature")
-        bpm, beats, beats_confidence, _, _ = rhythm_extractor(y)
+        estimator = BeatNet(
+            1,
+            mode="offline",
+            inference_model="DBN",
+            plot=[],
+            thread=False,
+        )
+        beat_frames = estimator.process(filename)
+        # result is a vector including beat times and downbeat identifier columns
+        # respectively with the following shape: numpy_array(num_beats, 2)
+        beat_frames = np.transpose(beat_frames)[0]
+
+        # compute bpm
+        if len(beat_frames) < 2:
+            return 0  # Not enough beats to calculate BPM
+
+        # Calculate intervals between beats
+        intervals = [
+            beat_frames[i] - beat_frames[i - 1] for i in range(1, len(beat_frames))]
+
+        # Average interval
+        avg_interval = sum(intervals) / len(intervals)
+
+        # Convert to BPM (60 seconds per minute)
+        tempo = 60 / avg_interval
+
+        return tempo, beat_frames
+
+    # def extract_beat_essentia(self, y, sr):
```

Comment:
any reason to not remove this instead of just commenting it out?

Reply:
good point. will remove